src/psbt.h

```diff
@@ -672,6 +718,80 @@ struct PSBTInput
                     UnserializeFromVector(s, m_tap_merkle_root);
                     break;
                 }
+                case PSBT_IN_MUSIG2_PARTICIPANT_PUBKEYS:
+                {
+                    if (!key_lookup.emplace(key).second) {
+                        throw std::ios_base::failure("Duplicate Key, input participant pubkeys for an aggregate key already provided");
+                    } else if (key.size() != 34) {
+                        throw std::ios_base::failure("Input musig2 participants pubkeys aggregate key is not 34 bytes");
+                    }
+                    std::array<unsigned char, 33> agg_key_bytes;
+                    skey >> AsWritableBytes(Span{agg_key_bytes});
+                    CPubKey agg_key(agg_key_bytes);
+
+                    std::vector<CPubKey> participants;
+                    std::vector<unsigned char> val;
+                    s >> val;
+                    SpanReader s_val{val};
+                    while (!s_val.empty()) {
+                        std::array<unsigned char, 33> part_key_bytes;
+                        s_val >> AsWritableBytes(Span{part_key_bytes});
```

Comment:
433eca29933447b70feba61fa26d3e6345fb3115: if `s_val` has less than 33 bytes remaining, this throws?

Reply:
Added a check for the size.

Reply:
Pfew. Separate from this PR I think it would be good if we had DSL that defines the PSBT spec and the constraints for every possible field. Maybe @ryanofsky has ideas?